prometheus-to-sd/config/source_config.go

```diff
@@ -112,7 +119,28 @@ func parseSourceConfig(uri flags.Uri, podId, namespaceId string) (*SourceConfig,
 	if err != nil {
 		return nil, err
 	}
-	return newSourceConfig(component, protocol, host, port, path, *auth, whitelisted, metricsPrefix, podConfig, whitelistedLabelsMap)
+	return newSourceConfig(component, protocol, host, port, path, *auth, whitelisted, metricsPrefix, podConfig, whitelistedLabelsMap, customResource, customLabels)
+}
+
+func getMap(v url.Values, name string) map[string]string {
+	n := len(name)
+	m := make(map[string]string)
+	for k, vals := range v {
+		if len(k) <= n {
```

Comment:
I believe it should be n+2. Otherwise passing "customLabels" as url query parameter will break the check below.

Reply:
Which check are you referring to? `k[n] == '['`? When `len(k) == n` (or less), we'll hit `continue` statement. When `len(k) == n+1`, both `k[:n]` and `k[n]` will work.